command-line/lib/Command/Address/Reward.hs

```diff
@@ -0,0 +1,90 @@
+    ( Cmd
+    , mod
+    , run
+    ) where
+
+import Prelude hiding
+    ( mod )
+
+import Cardano.Address
+    ( NetworkTag (..), bech32With )
+import Cardano.Address.Style.Shelley
+    ( mkNetworkDiscriminant, shelleyMainnet, shelleyTestnet )
+import Codec.Binary.Bech32
+    ( humanReadablePartFromText )
+import Control.Monad.Catch
+    ( throwM )
+import Fmt
+    ( build, fmt )
+import Options.Applicative
+    ( CommandFields, Mod, command, footerDoc, header, helper, info, progDesc )
+import Options.Applicative.Discrimination
+    ( networkTagOpt )
+import Options.Applicative.Help.Pretty
+    ( bold, indent, string, vsep )
+import Options.Applicative.Style
+    ( Style (..) )
+import System.Exit
+    ( die )
+import System.IO
+    ( stdin, stdout )
+import System.IO.Extra
+    ( hGetXPub, progName )
+
+import qualified Cardano.Address.Style.Shelley as Shelley
+import qualified Data.ByteString.Char8 as B8
+import qualified Data.Text.Encoding as T
+
+
+newtype Cmd = Cmd
+    { networkTag :: NetworkTag
+    } deriving (Show)
+
+mod :: (Cmd -> parent) -> Mod CommandFields parent
+mod liftCmd = command "stake" $
+    info (helper <*> fmap liftCmd parser) $ mempty
+        <> progDesc "Create a stake address"
+        <> header "Create a stake address \
+            \that references a staking key (1-1)."
+        <> footerDoc (Just $ vsep
+            [ string "The public key is read from stdin."
+            , string ""
+            , string "Example:"
+            , indent 2 $ bold $ string $ "$ "<>progName<>" recovery-phrase generate --size 15 \\"
+            , indent 4 $ bold $ string $ "| "<>progName<>" key from-recovery-phrase Shelley > root.prv"
+            , indent 2 $ string ""
+            , indent 2 $ bold $ string "$ cat root.prv \\"
+            , indent 4 $ bold $ string $ "| "<>progName<>" key child 1852H/1815H/0H/2/0 > stake.prv"
+            , indent 2 $ string ""
+            , indent 2 $ bold $ string "$ cat stake.prv \\"
+            , indent 4 $ bold $ string $ "| "<>progName<>" key public \\"
+            , indent 4 $ bold $ string $ "| "<>progName<>" address stake --network-tag 0"
+            , indent 2 $ string "stake1uqly0fjvrgguywze067gwhsexggtj8rrdnxczgp5vexe8zgxqns3g"
+            ])
+  where
+    parser = Cmd
+        <$> networkTagOpt Shelley
+
+run :: Cmd -> IO ()
+run Cmd{networkTag} = do
+    xpub <- hGetXPub stdin
+    hpr <- hprFromNetTag
+    case (mkNetworkDiscriminant . fromIntegral . unNetworkTag) networkTag of
+        Left e -> die (fmt $ build e)
+        Right discriminant -> do
+            let addr = Shelley.stakeAddress discriminant (Shelley.liftXPub xpub)
+            B8.hPutStr stdout $ T.encodeUtf8 $ bech32With hpr addr
+  where
+    hprFromText = either throwM pure . humanReadablePartFromText
+    hprFromNetTag
+        | shelleyMainnet == networkTag = hprFromText "stake"
+        | shelleyTestnet == networkTag = hprFromText "stake_test"
+        | otherwise = hprFromText "addr"
```

Comment:
`stake` would be a better default for values beyond `1`. Or, alternatively, we could change the current parser for the network tag option to fail if provided anything different than `0` or `1` and use instead named constant like `testnet` & `mainnet` (nicely done in a separate PR).